shared/src/main/java/com/couchbase/lite/Result.java

```diff
@@ -107,146 +107,164 @@ public int count() {
+        List<Object> array = new ArrayList<>();
+        for (int i = 0; i < count(); i++)
+            array.add(getObject(i));
+        return array;
     }
 
     //---------------------------------------------
     // implementation of ReadOnlyDictionaryInterface
     //---------------------------------------------
     @Override
     public List<String> getKeys() {
-        return null;
+        return new ArrayList<>(rs.getColumnNames().keySet());
     }
 
     @Override
     public Object getObject(String key) {
-        return null;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getObject(index) : null;
     }
 
     @Override
     public String getString(String key) {
-        return null;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getString(index) : null;
     }
 
     @Override
     public Number getNumber(String key) {
-        return null;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getNumber(index) : null;
     }
 
     @Override
     public int getInt(String key) {
-        return 0;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getInt(index) : 0;
     }
 
     @Override
     public long getLong(String key) {
-        return 0;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getLong(index) : 0L;
     }
 
     @Override
     public float getFloat(String key) {
-        return 0;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getFloat(index) : 0.0f;
     }
 
     @Override
     public double getDouble(String key) {
-        return 0;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getDouble(index) : 0.0;
     }
 
     @Override
     public boolean getBoolean(String key) {
-        return false;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getBoolean(index) : false;
     }
 
     @Override
     public Blob getBlob(String key) {
-        return null;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getBlob(index) : null;
     }
 
     @Override
     public Date getDate(String key) {
-        return null;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getDate(index) : null;
     }
 
     @Override
     public ReadOnlyArrayInterface getArray(String key) {
-        return null;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getArray(index) : null;
     }
 
     @Override
     public ReadOnlyDictionaryInterface getDictionary(String key) {
-        return null;
+        int index = indexForColumnName(key);
+        return index >= 0 ? getDictionary(index) : null;
     }
 
     @Override
     public Map<String, Object> toMap() {
-        return null;
+        Map<String, Object> dict = new HashMap<>();
+        for (String name : rs.getColumnNames().keySet()) {
+            dict.put(name, getObject(indexForColumnName(name)));
```

Comment:
toMap() should return only JSON based object. I think getObject() can return non JSON based objects.